Library/Formula/z3.rb

```diff
@@ -0,0 +1,33 @@
+class Z3 < Formula
+  desc "A high-performance theorem prover"
+  homepage "https://github.com/Z3Prover/z3"
+  url "https://github.com/Z3Prover/z3/archive/z3-4.4.0.tar.gz"
+  sha256 "65b72f9eb0af50949e504b47080fb3fc95f11c435633041d9a534473f3142cba"
+  head "https://github.com/Z3Prover/z3.git"
+  revision 1
```

Comment:
@DomT4 we cannot reset the revision here, otherwise users who already installed this formula will see formula's version gets downgraded.